core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/teach-oppia-modal.controller.ts

```diff
@@ -142,53 +141,55 @@ angular.module('oppia').controller('TeachOppiaModalController', [
     };
 
     $scope.confirmAnswerAssignment = function(answerIndex) {
-      var answer = $scope.unresolvedAnswers[answerIndex];
-      $scope.unresolvedAnswers.splice(answerIndex, 1);
-
-      var classificationType = (
-        answer.classificationResult.classificationCategorization);
-      var truncatedAnswer = $filter(
-        'truncateInputBasedOnInteractionAnswerType')(
-        answer.answer, interactionId, 12);
-      var successToast = (
-        'The answer ' + truncatedAnswer +
-        ' has been successfully trained.');
-
-      if (classificationType === DEFAULT_OUTCOME_CLASSIFICATION) {
-        TrainingDataService.associateWithDefaultResponse(
-          answer.answer);
+      var unresolvedAnswer = $scope.unresolvedAnswers[answerIndex];
+
+      if (unresolvedAnswer) {
```

Comment:
Ditto: Why do we need this check now? Did it not work earlier?

Reply:
I added this check because `answerIndex` can be a non-existed index and `unresolvedAnswer` might be `undefined`. I don't think it was throwing an error before but I just did a double-check.

Reply:
Hm, couple of questions:
1. Why is `confirmAnswerAssignment(..)` called when the passed index is invalid (ie. in what scenario does this happen)? IMO that action should not be allowed.
2. Why does it not throw an error in this case? Can you make sure that it throws an error in such cases? I think we need to be logging such cases.

Reply:
1. I agree with you, I don't see any scenario for this but I have a question: should we always enforce that the actions won't break the application through checks or is it ok to keep without checks if we're sure that is not an allowed action?
2. If the first question you see it's ok to use checks, yeah I can add error in these cases.

Reply:
The place where the call is made should make sure that arguments passed are valid. If they aren't, the method should not be called. So that means, error handling should be done there (and not inside the function being called). Does that make sense?

Reply:
Sure, I'll fix it.

Reply:
Done.